source/common/http/http2/codec_impl.h

```diff
@@ -84,6 +84,17 @@ class ConnectionImpl : public virtual Connection, Logger::Loggable<Logger::Id::h
   Protocol protocol() override { return Protocol::Http2; }
   void shutdownNotice() override;
   bool wantsToWrite() override { return nghttp2_session_want_write(session_); }
+  // Propogate network connection watermark events to each stream on the connection.
+  void onUnderlyingConnectionAboveWriteBufferHighWatermark() override {
```

Comment:
I kind of like everything passing through the connection manager as the authority of client side connections but maybe that's just me?